micro-rdk/src/common/provisioning/server.rs

```diff
@@ -445,20 +446,33 @@ where
     }
 }
 
+#[derive(Error, Debug)]
+pub enum WifiManagerError {
+    #[error("cannot assign to heapless string")]
+    HeaplessStringError,
```

Comment:
are we sure it will always be due to a `heapless` error? should we not propagate the underlying message somehow like in `OtherError`

Reply:
This error is only returned when essentially using the `TryFrom<&str>` trait for `heapless::String`. The corresponding `Error` type for that trait implementation is `()`, so the lack of an underlying message feels all but guaranteed